src/NCronJob/Registry/JobRun.cs

```diff
@@ -88,16 +140,21 @@ public void AddState(JobState state)
         States.Add(state);
         OnStateChanged?.Invoke(this, state);
     }
-    
+
     public void NotifyStateChange(JobStateType type, string message = "")
     {
-        if (CurrentState.Type == type)
+        if (CurrentState.Type == type || IsCompleted)
```

Comment:
Without this, a faulted job (for instance) would eventually be marked as Completed.